components/engine_traits/src/raft_engine.rs

```diff
@@ -138,6 +139,12 @@ pub trait RaftEngine: RaftEngineReadOnly + PerfContextExt + Clone + Sync + Send
     where
         F: FnMut(u64) -> std::result::Result<(), E>,
         E: From<Error>;
+
+    /// Indicate whether region states should be recovered from raftdb and
+    /// replay raft logs.
+    /// When kvdb's write-ahead-log is disabled, the sequence number of the last
+    /// boot time is saved.
+    fn put_recover_state(&self, seqno: u64) -> Result<()>;
```

Comment:
I think this is ambiguous. It can be interpreted as "put a state, which is used for recover", but it actually means "put a number that will make start recover states from raft db".

Reply:
I think there's too much information to fit in one name (i.e. recovery mode, seqno). And it can change in the future. Also I think it is still essentially "a state used for recover". How we interpret it is application-level.

Reply:
Then there should be a struct `RecoverState` instead of just a number.